src/main/java/com/sap/oss/phosphor/fosstars/data/release/info/ReleaseInfoLoader.java

```diff
@@ -0,0 +1,92 @@
+package com.sap.oss.phosphor.fosstars.data.release.info;
+
+import com.sap.oss.phosphor.fosstars.data.AbstractReleaseInfoLoader;
+import com.sap.oss.phosphor.fosstars.data.DataProvider;
+import com.sap.oss.phosphor.fosstars.data.github.GitHubDataFetcher;
+import com.sap.oss.phosphor.fosstars.data.github.ReleasesFromGitHub;
+import com.sap.oss.phosphor.fosstars.model.Artifact;
+import com.sap.oss.phosphor.fosstars.model.Value;
+import com.sap.oss.phosphor.fosstars.model.ValueSet;
+import com.sap.oss.phosphor.fosstars.model.release.info.MavenArtifact;
+import com.sap.oss.phosphor.fosstars.model.release.info.NpmArtifact;
+import com.sap.oss.phosphor.fosstars.model.subject.oss.GitHubProject;
+import com.sap.oss.phosphor.fosstars.model.value.ArtifactVersions;
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * This data provider tries to fill out the
+ * {@link com.sap.oss.phosphor.fosstars.model.feature.oss.OssFeatures#RELEASED_ARTIFACT_VERSIONS}
+ * feature. It is based on one of the following data providers:
+ * <ul>
+ * <li>{@link ReleaseInfoFromMaven}</li>
+ * <li>{@link ReleaseInfoFromNpm}</li>
+ * </ul>
+ * The data provider uses {@link ReleasesFromGitHub} as a fallback if
+ * {@link com.sap.oss.phosphor.fosstars.model.feature.oss.OssFeatures#RELEASED_ARTIFACT_VERSIONS}
+ * feature could not be filled up from one of the above-mentioned data providers.
+ */
+public class ReleaseInfoLoader extends AbstractReleaseInfoLoader<Artifact> {
+
+  /**
+   * A {@link ReleasesFromGitHub} data provider.
+   */
+  private final DataProvider gitHubReleaseInfoProvider;
+
+  /**
+   * Initializes a data provider.
+   *
+   * @param fetcher An interface to GitHub.
+   */
+  public ReleaseInfoLoader(GitHubDataFetcher fetcher) {
+    Objects.requireNonNull(fetcher, "Oh no! You gave me a null instead of a GitHub fetcher!");
+    gitHubReleaseInfoProvider = new ReleasesFromGitHub(fetcher);
+  }
+
+  @Override
+  public DataProvider<Artifact> update(Artifact artifact, ValueSet values) throws IOException {
+    DataProvider provider = providerFor(artifact);
+    provider.update(artifact, values);
+
+    for (Value<?> value : values) {
+      if (value.get() instanceof ArtifactVersions && !((ArtifactVersions) value.get()).empty()) {
+        break;
+      }
+
+      if (artifact.gitHubProject().isPresent()) {
+        doUpdate(artifact.gitHubProject().get(), values);
+        break;
+      }
+    }
+    return this;
+  }
+
+  /**
+   * Checks and updates the {@link ValueSet} if any release info is found in {@link GitHubProject}.
+   * 
+   * @param project A {@link GitHubProject}.
+   * @param values A {@link ValueSet}.
+   * @throws IOException If something went wrong.
+   */
+  void doUpdate(GitHubProject project, ValueSet values) throws IOException {
```

Comment:
This method looks redundant since it just calls `gitHubReleaseInfoProvider.update()` (the method doesn't seem to check anything what mentioned in the javadoc).

Reply:
I have updated the javadoc, But I need this method to help me with the test case

Reply:
How about mocking `gitHubReleaseInfoProvider`?